### ISSUE TITLE
Show location of number in call history

### ISSUE THREAD

**Issue by Rexadev** [FossifyOrg/Phone#39]
**Checklist**
 - [x] I made sure that there are no existing issues - open or closed - to which I could contribute my information.
 - [x]  I have read the FAQ and my problem isn't listed.
 - [x]  I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
 - [x]  This issue contains only one feature request.
 - [x]  I have read and understood the [contribution guidelines.](https://github.com/FossifyOrg/General-Discussion?tab=readme-ov-file#faq)

**Is your feature request related to a problem? Please describe.**
I would like to see the city, state, province, country of the numbers

**Describe the solution you'd like**

**Comment by Aga-C** [comment]
I found this library: https://github.com/google/libphonenumber/, but I'm not sure whether we can use this to have such detailed information.

**Comment by Rexadev** [comment]
Let us start with country
@Aga-C

**Comment by rsb-23** [comment]
We can display country name using country codes stored as dictionary / HashMap.

**Comment by naveensingh** [comment]
Yes, anything beyond the country might be hard to do without increasing the app size. Last I checked libphonenumber's metadata was around 15-20MB and even then we won't have proper info for each number.

**Comment by vertigo220** [comment]
If app size is a concern, there could be a companion app or separate ones, one with and one without this and perhaps other advanced features, essentially a full and lite version. Whenever I get a call from an unknown number, I always have to look it up to see who it is or, at least, where it is, and it would be nice to have this kind of info displayed automatically.

**Comment by inson1** [comment]
Does somebody knows how many different locations that library actually has? so we could know if its worth it or not?

**Comment by miapuffia** [comment]
Looks like additional size might be smaller than expected and it can get pretty detailed with the location. With an accurate phone number it could pinpoint a town with less than 7000 population. See that in the screenshot [here](https://github.com/FossifyOrg/Phone/pull/337)
